src/deferred.js

```diff
@@ -0,0 +1,21 @@
+
+var oldDeferred = jQuery.Deferred;
+
+jQuery.Deferred = function() {
+	var deferred = oldDeferred.apply( this, arguments );
+
+	// Don't add this method if the current jQuery doesn't provide it
```

Comment:
Overriding it for the warning. I didn't want to add `.pipe` if, for example, they were using jQuery 1.5 because `.pipe` wasn't there and maybe they're doing a feature detect or something. I suppose I could always add it but it seemed better to only shim it if the original `Deferred` didn't have it.

But talking to the rubber ducky now I realize that I'm not correctly covering the version 1.6 to 1.8 region where we had `.then` but it didn't return a new promise. In those cases we should use the existing `.pipe` which does return a new promise.